addons/mail/static/src/models/message_view.js

```diff
@@ -115,9 +135,29 @@ registerModel({
                 this.highlight();
                 this.update({ doHighlight: clear() });
             }
-            if (this.threadViewOwnerAsLastMessageView && this.component && this.component.isPartiallyVisible()) {
+            if (this.threadViewOwnerAsLastMessageView && this.isPartiallyVisible()) {
                 this.threadViewOwnerAsLastMessageView.handleVisibleMessage(this.message);
             }
+            if (this.prettyBodyRef.el && this.message.prettyBody !== this.lastPrettyBody) {
+                this.prettyBodyRef.el.innerHTML = this.message.prettyBody;
+                this.update({ lastPrettyBody: this.message.prettyBody });
+            }
+            if (!this.prettyBodyRef.el) {
+                this.update({ lastPrettyBody: clear() });
+            }
+            // Remove all readmore before if any before reinsert them with insertReadMoreLess.
+            // This is needed because insertReadMoreLess is working with direct DOM mutations
+            // which are not sync with Owl.
+            if (this.contentRef.el) {
+                for (const el of [...this.contentRef.el.querySelectorAll(':scope .o_Message_readMoreLess')]) {
```

Comment:
```suggestion
                for (const el of this.contentRef.el.querySelectorAll(':scope .o_Message_readMoreLess')) {
```

Reply:
`querySelectorAll` can return `null`, which is not iterable.
Also it returns a `NodeList`, which is not an `Array`. So I'm not even sure `for of` is supported on `NodeList`, hence why casting into an `Array`.

Reply:
If you can spread it you can iterate it using `for...of` loops, both depending on `Symbol.iterator`.
> querySelectorAll can return null, which is not iterable.

AFAIK `querySelectorAll` does not return `null` but an empty `NodeList` if there is no match for the given selector. And even if it returned `null`, `null` is not an iterable as you already pointed out, so attempting to spread it would result in an error either way.

Reply:
Hm... Indeed `querySelectorAll` always return `NodeList`, and apparently `for of` works fine on `NodeList`.
Will make the change then. Thanks!

Reply:
Change done